src/content/core/okteto-variables.mdx

```diff
@@ -7,36 +7,259 @@ id: okteto-variables
+    - echo "Running tests against ${SERVICE_ENDPOINT}..."
+```
+
+Another example for the same functionality, can be for example, creating the infrastructure, generating a `.env` file, and then deploy the application.
+
+```yaml
+deploy:
+  commands:
+  - name: Create AWS infrastructure
+    command: |
+      set -e
+      bucketName="${OKTETO_NAMESPACE}-okteto-tacos-shop"
+      terraform apply -input=false -var "bucket_name=$bucketName" -auto-approve
+      echo "S3_BUCKET_NAME=$resourceName" >> "$OKTETO_ENV"
+
+  - name: Create .env
+    command: |
+      set -e
+      echo "S3_BUCKET_NAME=$S3_BUCKET_NAME" > .env
+
+  - name: Deploy the app
+    command: |
+      set -e
+      make deploy
+```
+
+:::note
+
+Here are some highlights, and clarifications on the example above:
+
+1. The first step makes the variable `S3_BUCKET_NAME` available to the rest of the deployment steps. Both the step "Create .env" and "Deploy the app" will have access to `S3_BUCKET_NAME`.
+
+2. Also, it's important to remember that the variables in `$OKTETO_ENV` are not shared across `deploy`, `test`, and `destroy` sections. They are only shared within the same section.
+
+3. If you are wondering where the variable `OKTETO_NAMESPACE` comes from, check the section Okteto Variables [Managed by Okteto](#managed-by-okteto).
+
+:::
+
+
+### Configuring dynamic endpoints
+
+In this example, we show how `$OKTETO_ENV` can be used to create dynamic endpoints for external resources within your dev environment.
+
+If you are not familiar yet with the concept of Okteto external resource, image a component that is not part of the deployment of your dev environment, but that you need to interact with. For example, a database, a message queue, or an S3 bucket. If you need more details on this feature, check out [External Resources](/docs/development/external-resources).
+
+External resources are defined in the Okteto manifest and have a name and a URL. We understand that often, when using cloud providers, the resource URLs are generated dynamically when the infrastructure is provisioned. In this example, we make the S3 bucket name available to the rest of the deployment steps.
+
+```yaml
+## TODO: write example
+```
+
+
+## Setting Okteto Variables
+
+This section will show a brief example of how to set each type of Okteto Variable.
+
+### The --var flag
+
+The `--var` flag is useful to specify ad-hoc variables when running a command like [deploy](/docs/reference/okteto-cli#deploy) or [preview deploy](/docs/reference/okteto-cli#deploy-3)
+
+Example:
+
+```console
+$ okteto deploy --var PORT=4000
+```
+
+```yaml
+deploy:
+  image: "node:${TAG}"
+  commands:
+    - name: deploy
+      command: NODE_ENV=${ENV} make deploy
+```
+
+### Catalog
+
+When you deploy an application from the Okteto Dashboard or if your organization is using the Okteto Catalog, you can define project-specific variables. See [Catalog Variables](/docs/admin/catalog/#add-environment-variables) for more information.
```

Comment:
```suggestion
When you deploy an application from the Okteto UI or if your organization uses the Okteto Catalog, you can define variables specific for your development environment. See [Catalog Variables](/docs/admin/catalog/#add-environment-variables) for more information.
```
We don't use the term "project" in our docs. I know it's hard, and that our terms might not be ideal, but it's better to stick to them than adding new synonymous